### DISCUSSION TITLE
How to use match with two different bundle ids?

### DISCUSSION THREAD

**Discussion by AdamGerthel** [fastlane/fastlane#17403] [Q&A] [answered]
I'm currently about to go insane trying to understand how to use match with two different bundle ids. Basically, I have one bundle ID for development, and another for production. I am able to build these using Fastlane **if** I change the Provisioning Profile value for "Signing (Release)" within xcode. But - afaik match is suppose to handle this for me.

I've read the [match documentation](https://docs.fastlane.tools/actions/match/) as well as [Setting up your Xcode Project](https://docs.fastlane.tools/codesigning/xcode-project/) from top to bottom but I can't for the life of me understand what I'm doing wrong.

The error I keep getting (unless I've manually changed to the correct value in xcode) is of this type:

> error: Provisioning profile "match AppStore com.xxx.xxx.ios" has app ID "com.xxx.xxx.ios", which does not match the bundle ID "com.xxx.xxx.dev". (in target 'example' from project 'example')

But as you can see in the screenshot below, the Gym summary clearly (at least to me) states that the profile matching is correct:

![Screenshot 2020-10-07 at 16 59 49](https://user-images.githubusercontent.com/1490370/95349775-bbf36b80-08bf-11eb-9819-cde1ce682d21.png)

My complete fastfile contains some unrelated stuff as well, but this is the gist of my current build setup:

```
project_name = "untold"
xcodeproj = "./ios/#{project_name}.xcodeproj"
workspace = "./ios/#{project_name}.xcworkspace"
ios_build_directory = "./ios/builds"
build_type = "Release"
type = "appstore"

match(type: type)

update_app_identifier(
  xcodeproj: xcodeproj,
  plist_path: "#{project_name}/Info.plist",
  app_identifier: CredentialsManager::AppfileConfig.try_fetch_value(:app_identifier),
)

update_project_team(
  teamid: CredentialsManager::AppfileConfig.try_fetch_value(:team_id),
  path: xcodeproj
)

build_app(
  workspace: workspace,
  scheme: project_name,
  clean: true,
  configuration: build_type,
  build_path: ios_build_directory,
  output_directory: ios_build_directory,
  output_name: "#{project_name}.ipa",
  export_options: {
    provisioningProfiles: lane_context[SharedValues::MATCH_PROVISIONING_PROFILE_MAPPING],
  }
)
```
FYI, I use different env variables to set different app identifiers for the the app versions.

I've tried without `export_options` as well, since this is never mentioned in any of the match. I've also tried with/without `skip_profile_detection` but the problem remains.

What am I doing wrong? Isn't match supposed to automatically handle this?

**Comment by rogerluan** [comment]
Hey @AdamGerthel 👋 
I may be wrong here, but I think _match_ acts on a single app at once only. So if you want it to act on 2 apps (2 bundle IDs), you need to call it twice, explicitly passing the bundle IDs as arguments instead of using global env vars for this. 

Does this help? 🙈

**Reply by AdamGerthel** [reply]
> I may be wrong here, but I think match acts on a single app at once only.

Well, maybe it does.. but I'm doing the build one at a time, and my match git repository has profiles for both bundle ids. Neither match nor xcode should really need to be aware of the fact that I do have multiple, since I use the `app_identifier`.

If I update `PROVISIONING_PROFILE_SPECIFIER` in xcode manually to the proper value, then the lane works. I would've assumed that match would handle that for me, or at least that that's what export_options in build_app allows me to set. 

In any case, per your suggestion, I tried specifying the bundle id specifically for match:
```
match(type: "appstore", app_identifier: CredentialsManager::AppfileConfig.try_fetch_value(:app_identifier))
```
But I got the same exact result. This is not unexpected since match already reads the app_identifier value.

**Comment by AdamGerthel** [comment, marked as answer]
My issue seems to be related to this: https://github.com/fastlane/fastlane/issues/15338

But there's no solution mentioned there which will work for me since I can't use environment variables which use bundle ids.

UPDATE: I think I solved it by using `update_project_provisioning` before running `build_app` with the profile path supplied by `match` (which uses `sigh`)

```
    update_project_provisioning(
      xcodeproj: xcodeproj,
      profile: ENV["sigh_#{app_identifier}_appstore_profile-path"],
      build_configuration: "Release"
    )
```
